### DISCUSSION TITLE
Handle unexpected args while using pydantic?

### DISCUSSION THREAD

**Discussion by Leezj9671** [fastapi/fastapi#7153] [Questions] [answered]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [ ] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [ ] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [ ] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from typing import Optional

from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI()

class Item(BaseModel):
    name: str
    price: float

@app.post("/items/{item_id}")
def update_item(item_id: int, item: Item):
    requests.post(another_api, item) # example
    return {"item_name": item.name, "item_id": item_id, "item": item}
```

### Description

I want to hadnle some unexpected args, and forward these args to other API. For example, my post body as below:

```json
{
    "name": "testman",
    "price": 1.1,
    "this_is_other_args": "example_values"
}
```

Because of pydantic `BaseModel`, I got `item` in `update_item` without `this_is_other_args`. How can I handle this args like `**kwargs` in python?

**Comment by daviskirk** [comment, marked as answer]
You can use pydantics model "Config" to do this:

```python
from fastapi import FastAPI
from pydantic import BaseModel


app = FastAPI()

class Item(BaseModel):
    name: str
    price: float

    class Config:
        extra = 'allow'  # anything can be assigned to the item, even if not specified as a field


@app.post("/items/{item_id}")
def update_item(item_id: int, item: Item):
    # extras would hold any field that isnt name or price
    extras = item.dict(exclude=item.__fields__.keys())
    print(f'kwargs: {extras}')
    # only include the models fields
    real_stuff = item.dict(include=item.__fields__.keys())
    print(f'item: {real_stuff}'
```
 
The out put of: 

```python
requests.post('http://127.0.0.1:8000/items/1', json = {
    "name": "testman",
    "price": 1.1,
    "this_is_other_args": "example_values"
})
```

then is

```
kwargs: {'this_is_other_args': 'example_values'}
item: {'name': 'testman', 'price': 1.1}
```

see:

- https://pydantic-docs.helpmanual.io/usage/model_config/
- https://pydantic-docs.helpmanual.io/usage/exporting_models/#modeldict

**Comment by Leezj9671** [comment]
@daviskirk It prefectly solved this problem, and thx a lot!

**Comment by tiangolo** [comment]
Thanks for the help here @daviskirk! 🍰 

And thanks for coming back to close the issue @Leezj9671 ☕ 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
